### PR TITLE
Rename storage.Key to storage.Address

### PR DESCRIPTION
**Pull request by gbalint**
First trivial PR to split the chunk refactor code to smaller steps

### CODE REVIEW

**Review by janos — COMMENTED** [review]
There are other variables and functions that contain Key in name. It would be much bigger pr, so this can be left for later to change, if needed.

**Comment by nonsense** [comment]
LGTM, but I agree with @janos . Maybe we can include at least the `key storage.Key` rename to `address storage.Address`, instead of `key storage.Address`. The rest we could probably refactor as we go.

**Comment by gbalint** [comment]
@nonsense @janos I added a huge bunch of renames in 8de453dee82d9d864c89a2d55a058a49e1ac3118

File: swarm/api/api.go [hunk 1]
```diff
@@ -310,7 +310,7 @@ func (self *Api) Put(content, contentType string, toEncrypt bool) (k storage.Key
 // Get uses iterative manifest retrieval and prefix matching
 // to resolve basePath to content using dpa retrieve
 // it returns a section reader, mimeType, status, the key of the actual content and an error
-func (self *Api) Get(manifestKey storage.Key, path string) (reader storage.LazySectionReader, mimeType string, status int, contentKey storage.Key, err error) {
+func (self *Api) Get(manifestKey storage.Address, path string) (reader storage.LazySectionReader, mimeType string, status int, contentKey storage.Address, err error) {
```
Comment (janos) [inline swarm/api/api.go]: To rename it to manifestAddr?

File: swarm/api/http/server.go [hunk 2]
```diff
@@ -626,9 +626,9 @@ func (s *Server) HandleGet(w http.ResponseWriter, r *Request) {
 	log.Debug("handle.get", "ruid", r.ruid, "uri", r.uri)
 	getCount.Inc(1)
 	var err error
-	key := r.uri.Key()
-	if key == nil {
-		key, err = s.api.Resolve(r.uri)
+	addr := r.uri.Key()
```
Comment (janos) [inline swarm/api/http/server.go]: Should `URI.Key()` be renamed to `URI.Address()`? BTW, URI type already contains `addr` and `Addr`, but fortunately with nice comments.

File: swarm/api/http/server.go [hunk 3]
```diff
@@ -180,26 +180,26 @@ func (s *Server) HandlePostFiles(w http.ResponseWriter, r *Request) {
...
 			postFilesFail.Inc(1)
 			Respond(w, r, err.Error(), http.StatusInternalServerError)
 			return
 		}
-		log.Debug("new manifest", "ruid", r.ruid, "key", key)
+		log.Debug("new manifest", "ruid", r.ruid, "key", addr)
 	}
 
-	newKey, err := s.updateManifest(key, func(mw *api.ManifestWriter) error {
+	newKey, err := s.updateManifest(addr, func(mw *api.ManifestWriter) error {
```
Comment (janos) [inline swarm/api/http/server.go]: newAddr?

**Review by nonsense — APPROVED** [review]
LGTM

File: swarm/network/stream/delivery.go [hunk 4]
```diff
@@ -131,7 +131,7 @@ func (s *SwarmChunkServer) GetData(key []byte) ([]byte, error) {
 
 // RetrieveRequestMsg is the protocol msg for chunk retrieve requests
 type RetrieveRequestMsg struct {
-	Key       storage.Key
+	Key       storage.Address
```
Comment (janos) [inline swarm/network/stream/delivery.go]: Renaming field `Key` to `Address` will break backward compatibility. If we do not care for it currently, this should be also renamed. Like in other message types in this package.

**Review by janos — COMMENTED** [review]
There are more variables named key or newKey in swarm/api/http/server.go, for example in HandleDelete, handleDirectUpload... Also, comments are using word key instead of addr or address.

File swarm/api/api.go has variables named with word key in them like manifestKey, fkey, newMkey, key...

File: swarm/network/stream/delivery.go (hunk 4 above)
Comment (zelig) [reply swarm/network/stream/delivery.go]: i dont think it necessarily break anything actually. we use RLP encoding using reflect
if reflect's field order is unchanged, there is no problem

Comment (janos) [reply swarm/network/stream/delivery.go]: That is great. 👍

**Comment by gbalint** [comment]
@janos I added more renames according to your comments and further

**Pull request merged**